kernel/global-clock/core/src/main/java/org/apache/shardingsphere/globalclock/core/rule/GlobalClockRule.java

```diff
@@ -40,12 +46,26 @@ public final class GlobalClockRule implements GlobalRule {
     
     private final boolean enabled;
     
-    public GlobalClockRule(final GlobalClockRuleConfiguration ruleConfig) {
+    public GlobalClockRule(final GlobalClockRuleConfiguration ruleConfig, final Map<String, ShardingSphereDatabase> databases) {
         configuration = ruleConfig;
         enabled = ruleConfig.isEnabled();
         globalClockProvider = enabled ? TypedSPILoader.getService(GlobalClockProvider.class, String.join(".", ruleConfig.getType(), ruleConfig.getProvider()),
                 null == ruleConfig.getProps() ? new Properties() : ruleConfig.getProps()) : null;
-        Optional.ofNullable(globalClockProvider).ifPresent(optional -> optional.init(ruleConfig.getProps()));
+        TypedSPILoader.getService(TransactionHook.class, "GlobalClock", getProps(ruleConfig, databases));
+    }
+    
+    private Properties getProps(final GlobalClockRuleConfiguration ruleConfig, final Map<String, ShardingSphereDatabase> databases) {
+        Properties result = new Properties();
```

Comment:
You can use PropertiesBuilder to reduce code.

Reply:
PropertiesBuilder just in test scope now.